controllers/argocd/server/clusterrole.go

```diff
@@ -0,0 +1,127 @@
+package server
+
+import (
+	"reflect"
+
+	"github.com/argoproj-labs/argocd-operator/common"
+	"github.com/argoproj-labs/argocd-operator/pkg/mutation"
+	"github.com/argoproj-labs/argocd-operator/pkg/permissions"
+	v1 "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (sr *ServerReconciler) reconcileClusterRole() error {
+
+	crName := getClusterRoleName(sr.Instance.Name, sr.Instance.Namespace)
+	crLables := common.DefaultResourceLabels(crName, sr.Instance.Name, ServerControllerComponent)
```

Comment:
We should generate all the resource names we need in `server.go` before the reconciliation begins, and just use those values wherever applicable, so that we only define them once and it stays consistent across all resources

labels will be generated by `argoutil.GetObjMeta` so we don't need to worry about those here